common/components/AddressFieldFactory/AddressInputFactory.tsx

```diff
@@ -1,29 +1,51 @@
 import React, { Component } from 'react';
-import { Identicon } from 'components/ui';
+import { Identicon, Spinner } from 'components/ui';
 import translate from 'translations';
-//import { EnsAddress } from './components';
 import { Query } from 'components/renderCbs';
 import { ICurrentTo, getCurrentTo, isValidCurrentTo } from 'selectors/transaction';
 import { connect } from 'react-redux';
 import { AppState } from 'reducers';
 import { CallbackProps } from 'components/AddressFieldFactory';
+import { addHexPrefix } from 'ethereumjs-util';
+import { getResolvingDomain } from 'selectors/ens';
+import { isValidENSAddress } from 'libs/validators';
 
 interface StateProps {
   currentTo: ICurrentTo;
   isValid: boolean;
+  isResolving: boolean;
 }
+
 interface OwnProps {
   onChange(ev: React.FormEvent<HTMLInputElement>): void;
   withProps(props: CallbackProps): React.ReactElement<any> | null;
 }
 
+const ENSStatus: React.SFC<{ isLoading: boolean; ensAddress: string; rawAddress: string }> = ({
+  isLoading,
+  ensAddress,
+  rawAddress
+}) => {
+  const isENS = isValidENSAddress(ensAddress);
+  const text = 'Loading ENS address...';
+  if (isLoading) {
+    return (
+      <>
+        <Spinner /> {text}
+      </>
+    );
+  } else {
+    return isENS ? <>{`Resolved Address: ${rawAddress}`}</> : null;
+  }
```

Comment:
The loader shows up even if the address is invalid (i.e. too short) but then you get no message explaining what happened. There should be a message for "invalid ens".

<img width="622" alt="screen shot 2018-01-15 at 1 20 49 am" src="https://user-images.githubusercontent.com/649992/34929521-5e4d8a86-f992-11e7-9e66-2a6f9bcdcdb2.png">